platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/Wire.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring;
+
+import com.swirlds.common.wiring.counters.ObjectCounter;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.function.Consumer;
+
+/**
+ * Wires two or more components together.
+ *
+ * @param <O> the output time of the wire (use {@link Void}) for a wire with no output type)
+ */
+public abstract class Wire<O> extends OutputChannel<O> {
+
+    private final boolean flushEnabled;
+
+    /**
+     * Constructor.
+     *
+     * @param model               the wiring model containing this wire
+     * @param name                the name of the wire
+     * @param flushEnabled        if true, then {@link #flush()} and {@link #interruptableFlush()} will be enabled,
+     *                            otherwise they will throw.
+     * @param insertionIsBlocking when data is inserted into this wire, will it block until capacity is available?
+     */
+    protected Wire(
+            @NonNull final WiringModel model,
+            @NonNull final String name,
+            final boolean flushEnabled,
+            final boolean insertionIsBlocking) {
+        super(model, name, insertionIsBlocking);
+        this.flushEnabled = flushEnabled;
+    }
+
+    /**
+     * Cast this wire into whatever a variable is expecting. Sometimes the compiler gets confused with generics, and
+     * path of least resistance is to just cast to the proper data type.
+     *
+     * <p>
+     * Warning: this will appease the compiler, but it is possible to cast a wire into a data type that will cause
+     * runtime exceptions. Use with appropriate caution.
+     *
+     * @param <X> the type to cast to
+     * @return this, cast into whatever type is requested
+     */
+    @NonNull
+    @SuppressWarnings("unchecked")
+    public final <X> Wire<X> cast() {
```

Comment:
I'm not sure this is needed

Reply:
This is actually super useful, ping me and I'll show you examples.